impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -492,32 +484,59 @@ private List<String> buildScramArguments() {
                 disableControllerCheck(zkClient);
                 uscrs.forEach(credentials -> {
-                    var userConfig = adminZkClient.fetchEntityConfig(userEntityType, credentials.name());
+                    var userConfig = fetchEntityConfig(adminZkClient, userEntityType, credentials.name());
                     var credentialsString = ScramCredentialUtils.credentialToString(ScramUtils.asScramCredential(credentials));
 
                     userConfig.setProperty(ScramMechanism.fromType(credentials.mechanism()).mechanismName(), credentialsString);
-                    adminZkClient.changeConfigs(userEntityType, credentials.name(), userConfig, false);
+                    changeConfigs(adminZkClient, userEntityType, credentials.name(), userConfig, false);
                 });
             }
+            catch (Exception e) {
+                throw new IllegalStateException("Failed to create scram users", e);
+            }
         }
     }
 
+    private void changeConfigs(Object adminZkClient, String userEntityType, String name, Properties userConfig, boolean isUserClientId) {
+        ReflectionUtils.invokeInstanceMethod(adminZkClient, "changeConfigs", userEntityType, name, userConfig, isUserClientId);
+    }
+
+    private Properties fetchEntityConfig(Object adminZkClient, String userEntityType, String name) {
+        return ReflectionUtils.invokeInstanceMethod(adminZkClient, "fetchEntityConfig", userEntityType, name);
+    }
+
+    private AutoCloseable createZkClient(String name, Time system, KafkaConfig config, ZKClientConfig zkClientConfig) {
+
+        try {
+            var clazz = Class.forName("kafka.zk.KafkaZkClient");
+            return ReflectionUtils.invokeStaticMethod(clazz, "createZkClient", name, system, config, zkClientConfig);
+        }
+        catch (Exception e) {
+            throw new IllegalStateException("Failed to invoke createZkClient while creating scram users", e);
+        }
+    }
+
+    private ZKClientConfig zkClientConfigFromKafkaConfig(Class<Server> kafkaServerClazz, KafkaConfig config) {
+        return ReflectionUtils.invokeStaticMethod(kafkaServerClazz, "zkClientConfigFromKafkaConfig", config, false);
+    }
+
     // The accessibility hack is tactical for Kafka >=3.7.1 <= 4.0.0
     // The alternative is copying the code that constructs the ZK client see https://github.com/ozangunalp/kafka-native/pull/195/files as an example. Both options suck,
     // this is less code, and thus we hope less likely to bit rot.
     @SuppressWarnings("java:S3011")
-    private static void disableControllerCheck(KafkaZkClient zkClient) {
+    private static void disableControllerCheck(Object zkClient) {
         try {
-            final Class<? extends KafkaZkClient> zkClientClass = zkClient.getClass();
+            final Class<?> zkClientClass = zkClient.getClass();
             final Field enableEntityConfigControllerCheck = zkClientClass.getDeclaredField("enableEntityConfigControllerCheck");
```

Comment:
I think we should shuffle the field manipulation to ReflectionUtils too

Reply:
A neat abstraction isn't really leaping out to me.  We'd need the API to throw the NoSuchFieldException so we can cater for the case where the we are on <= 3.7.0.    I'll leave this for now